pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -292,12 +292,12 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
                 tableNameWithType);
         zkDownloadUri = downloadUri;
       } else {
-        zkDownloadUri = getZkDownloadURIForSegmentUpload(tableNameWithType, segmentName);
+        zkDownloadUri = getZkDownloadURIForSegmentUpload(tableName, segmentName);
       }
 
       // Zk operations
-      completeZkOperations(enableParallelPushProtection, headers, finalSegmentFile, tableNameWithType, segmentMetadata,
-          segmentName, zkDownloadUri, moveSegmentToFinalLocation, crypterClassName);
+      completeZkOperations(enableParallelPushProtection, headers, finalSegmentFile, tableName, tableNameWithType,
```

Comment:
Passing both `tableNameWithType` and `tableName` seems redundant. If we pass `tableNameWithType`, we should be able to derive `tableName` inside `completeZkOperations()` method, right?

Reply:
I originally had it like that and derived rawTableName by invoking `TableNameBuilder.extractRawTableName()`. But `completedZkOperations()` is a private method that's being used only once.  We already have rawTableName in the caller, so we can pass it there as well to prevent another function call to extract the raw table name.
If you still think it looks better to pass only tableNameWithType, I can simply change it.

Reply:
I think it is better to reduce the number of arguments, yes.